Doc enhancements for on-prem deployments (#5176)
pachyderm/pachyderm: doc/docs/1.11.x/deploy-manage/deploy/on_premises.md

For an on-premises deployment, 
it's not advisable to use a cloud-based storage mechanism.

> Don't deploy an on-premises Pachyderm cluster against cloud-based object stores such as S3 from [AWS](amazon_web_services/index.md), GCS from [Google Cloud Platform](google_cloud_platform.md), Azure Blob Storage from [Azure](azure.md). Don't get confused that the command line parameters for the object store (`--object-store`) are specifying `s3` - in this case, the argument is referring to the S3 protocol (which is also used by solutions such as MinIO and the likes) and not the Amazon product with the same name.

Reviewer: ```suggestion
Don't deploy an on-premises Pachyderm cluster against cloud-based object stores such as S3 from [AWS](amazon_web_services/index.md), GCS from [Google Cloud Platform](google_cloud_platform.md), Azure Blob Storage from [Azure](azure.md). Note that the command line parameters for the object store (`--object-store`) are specifying `s3` in reference to the S3 protocol (which is used by solutions such as MinIO and the like) and not the Amazon product with the same name.
```

1. `OS_ENDPOINT`: The access endpoint.
   For example, MinIO's endpoints are usually something like `minio-server:9000`. 

>    Don't begin it with the protocol; it's an endpoint, not an url. Also, check if your local instance of e.g. MinIO is using SSL/TLS

Reviewer: By local instance, are you referring to the client running on the local machine?

Author: "Local" is of course not very precise, well spotted. I was referring to the on-prem deployment of MinIO in the k8s cluster. A better formulation might be "Also, check if the object storage deployment (e.g. MinIO) in your cluster uses SSL/TLS ..."

>    Don't begin it with the protocol; it's an endpoint, not an url. Also, check if your local instance of e.g. MinIO is using SSL/TLS
>    and consider using `--disable-ssl` if not.

Reviewer: ```suggestion
   Don't begin it with the protocol; it's an endpoint, not an url. Also, check if your object store (e.g. MinIO) is using SSL/TLS.
   If not, disable it using `--disable-ssl`.
```

Reviewer: Does this clear it up a bit or did I make it more complicated? hah
